sanic/models/futures.py

```diff
@@ -1,39 +1,46 @@
-from collections import namedtuple
-
-
-FutureRoute = namedtuple(
-    "FutureRoute",
-    [
-        "handler",
-        "uri",
-        "methods",
-        "host",
-        "strict_slashes",
-        "stream",
-        "version",
-        "name",
-        "ignore_body",
-        "websocket",
-        "subprotocols",
-        "unquote",
-        "static",
-    ],
-)
-FutureListener = namedtuple("FutureListener", ["listener", "event"])
-FutureMiddleware = namedtuple("FutureMiddleware", ["middleware", "attach_to"])
-FutureException = namedtuple("FutureException", ["handler", "exceptions"])
-FutureStatic = namedtuple(
-    "FutureStatic",
-    [
-        "uri",
-        "file_or_directory",
-        "pattern",
-        "use_modified_since",
-        "use_content_range",
-        "stream_large_files",
-        "name",
-        "host",
-        "strict_slashes",
-        "content_type",
-    ],
-)
+from pathlib import PurePath
+from typing import NamedTuple, List, Union, Coroutine, Any
+
```

Comment:
Hope you don't mind the rewrite of this entire file. Makes it much easier to auto complete and lint things this way.

Reply:
Works for me. Was on the mental list of things to clean up.